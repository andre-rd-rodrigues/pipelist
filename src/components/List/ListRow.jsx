import React from "react";
import OfficeIcon from "assets/icons/office_icon";
import Avatar from "components/Avatar/Avatar";
import styles from "./list.module.scss";

const ListRow = ({ onPersonSelect, person, provided }) => {
  const onRowClicked = () => {
    onPersonSelect(person);
  };
  return (
    <div
      className={styles.listRow}
      role="button"
      onClick={onRowClicked}
      onKeyPress={onRowClicked}
      tabIndex={0}
      ref={provided.innerRef}
      {...provided.draggableProps}
      {...provided.dragHandleProps}
    >
      <div>
        <h4>{`${person?.first_name} ${person?.last_name}`}</h4>
        <div className="list_row_person_address_div">
          <OfficeIcon />
          <p>{person?.org_name}</p>
        </div>
      </div>
      <Avatar src={person?.picture_id?.pictures[128]} alt="Pipelist avatar" />
    </div>
  );
};

export default ListRow;
