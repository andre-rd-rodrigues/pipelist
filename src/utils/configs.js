const data_fields = {
  groups: "f7cf8b3ea06ed72293f69d25fd6406079b1cedfe",
  assistant: "1102c3288d42c7ae3b1faf65a80a51521337af06",
  location: "e985f13166eabac0eeef823f9de603a4159952da"
};

export { data_fields };
