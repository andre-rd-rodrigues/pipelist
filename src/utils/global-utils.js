import Notification from "components/Notification/Notification";

const notificationReload = (message, type) => {
  Notification(message, type);
  return setTimeout(() => {
    window.location.reload();
  }, 1800);
};

export { notificationReload };
