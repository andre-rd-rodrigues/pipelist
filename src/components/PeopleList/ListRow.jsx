import React from "react";
import OfficeIcon from "assets/icons/office_icon";
import Avatar from "components/Avatar/Avatar";
import { empty_field } from "utils/configs";
import styles from "./list.module.scss";

const ListRow = ({ onPersonSelect, person, provided }) => {
  const { innerRef, draggableProps, dragHandleProps } = provided;

  const onRowClicked = () => {
    onPersonSelect(person);
  };
  return (
    <div
      className={styles.listRow}
      role="button"
      onClick={onRowClicked}
      onKeyPress={onRowClicked}
      tabIndex={0}
      ref={innerRef}
      {...draggableProps}
      {...dragHandleProps}
      data-testid="list_row_div"
    >
      <div>
        <h4>{`${person?.first_name} ${person?.last_name}`}</h4>
        <div className="list_row_person_address_div">
          <OfficeIcon />
          <p>{person?.org_name || empty_field}</p>
        </div>
      </div>
      <Avatar src={person?.picture_id?.pictures[128]} alt="Pipelist avatar" />
    </div>
  );
};

export default ListRow;
