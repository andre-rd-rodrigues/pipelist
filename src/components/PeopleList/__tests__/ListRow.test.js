import { fireEvent, render, screen } from "@testing-library/react";
import reactTestRenderer from "react-test-renderer";
import { LoadingProviderMock, mocked_list } from "utils/test-utils";
import ListRow from "../ListRow";

describe("Loader", () => {
  let props, person;

  person = mocked_list[0];
  props = {
    innerRef: jest.fn(),
    provided: {
      innerRef: jest.fn(),
      draggableProps: {},
      dragHandleProps: {}
    },
    onPersonSelect: jest.fn(),
    person
  };

  it("should render correctly", () => {
    const wrapper = reactTestRenderer.create(
      <LoadingProviderMock>
        <ListRow {...props} />
      </LoadingProviderMock>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render according to props", () => {
    let newProps = {
      ...props,
      person: {
        ...mocked_list[0],
        org_name: "Andre Pharmacy"
      }
    };
    render(
      <LoadingProviderMock>
        <ListRow {...newProps} />
      </LoadingProviderMock>
    );
    expect(screen.getByText(/Andre Pharmacy/i)).toBeInTheDocument();
  });

  it("should run onPersonSelect when div[role='button'] clicked", () => {
    render(
      <LoadingProviderMock>
        <ListRow {...props} />
      </LoadingProviderMock>
    );

    fireEvent.click(screen.getByTestId("list_row_div"));

    expect(props.onPersonSelect).toBeCalled();
  });
});
