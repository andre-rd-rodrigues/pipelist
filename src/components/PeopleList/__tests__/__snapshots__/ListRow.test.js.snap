// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly 1`] = `
<div
  className="mb-4"
>
  <div
    className="listRow"
    data-testid="list_row_div"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={Object {}}
    tabIndex={0}
  >
    <div>
      <h4>
        Mila Benson
      </h4>
      <div
        className="list_row_person_address_div"
      >
        <svg
          enableBackground="new 0 0 24 24"
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <rect
              fill="none"
              height={24}
              width={24}
            />
          </g>
          <g>
            <path
              d="M17,11V3H7v4H3v14h8v-4h2v4h8V11H17z M7,19H5v-2h2V19z M7,15H5v-2h2V15z M7,11H5V9h2V11z M11,15H9v-2h2V15z M11,11H9V9h2 V11z M11,7H9V5h2V7z M15,15h-2v-2h2V15z M15,11h-2V9h2V11z M15,7h-2V5h2V7z M19,19h-2v-2h2V19z M19,15h-2v-2h2V15z"
              fill="currentColor"
            />
          </g>
        </svg>
        <p>
          Rite Aid Pharmacy
        </p>
      </div>
    </div>
    <div
      className="avatar"
      id="app_avatar"
      style={
        Object {
          "backgroundImage": "url(https://pipedrive-profile-pics.s3.amazonaws.com/c662095da1a8169cb828f10740d0d69dba97aa5b_128.jpg)",
          "height": "4rem",
          "width": "4rem",
        }
      }
    />
  </div>
</div>
`;
